extensions/wikia/Email/Controller/WallMessageController.class.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace Email\Controller;
+
+use Email\EmailController;
+
+class WallMessageController extends EmailController {
+
+	/**
+	 * @template wallMessage
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'summary' => $this->getSummary(),
+			'editorProfilePage' => $this->getCurrentProfilePage(),
+			'editorUserName' => $this->request->getVal( 'fromName' ),
+			'editorAvatarURL' => $this->getCurrentAvatarURL(),
+			'wallMessageTitle' => $this->request->getVal( 'messageTitle' ),
+			'wallMessageBody' => $this->request->getVal( 'messageBody' ),
+			'buttonText' => $this->getFullConversationText(),
+			'buttonLink' => $this->getFullConversationUrl(),
+			'contentFooterMessages' => [
+				$this->getRecentMessagesText()
+			],
+		] );
+	}
+
+	/**
+	 * Whether this email is for a user following a thread on someone else's wall
+	 *
+	 * @return boolean
+	 */
+	protected function isForFollowedThread() {
+		return $this->request->getVal( 'targetUser' ) !== $this->request->getVal( 'wallUserName' );
+	}
+
+	/**
+	 * Get the email subject line
+	 *
+	 * @return string
+	 */
+	public function getSubject() {
+		if ( $this->isForFollowedThread() ) {
+			return wfMessage( 'emailext-wallmessage-following-subject',
+				$this->request->getVal( 'fromName' ),
+				$this->request->getVal( 'wallUserName' )
```

Comment:
So it might be nice to validate these values before we get here. The pattern we've been kind of following is to put these into instance variables inside of initEmail() and then validate them there. That way we can throw an exception if we don't get passed the right values or they're invalid.

It doesn't really come up when the email is sent by the specified user action, but as we're testing emails/sending them manually that kind of feedback is really helpful. Like this: https://github.com/Wikia/app/blob/dev/extensions/wikia/Email/Controller/WatchedPageController.class.php#L26-L67